lib/screens/authenticate/verified_email_check.dart

```diff
@@ -0,0 +1,171 @@
+import 'dart:async' show Future, Timer;
+
+import 'package:flutter/material.dart';
+import 'package:shareacab/screens/rootscreen.dart';
+import 'package:shareacab/services/auth.dart';
+import 'package:provider/provider.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+
+class VerificationCheck extends StatefulWidget {
+  @override
+  _VerificationCheckState createState() => _VerificationCheckState();
+}
+
+class _VerificationCheckState extends State<VerificationCheck> {
+  final AuthService _auth = AuthService();
+  FirebaseAuth auth;
+  final _formKey = GlobalKey<FormState>();
+  bool loading = false;
+  String error = '';
+  bool verified = false;
+
+  // void _checkIfVerified() async {
+  //   await FirebaseAuth.instance.currentUser()
+  //     ..reload();
+  //   var user = await FirebaseAuth.instance.currentUser();
+  //   if (user.isEmailVerified) {
+  //     setState(() {
+  //       verified = user.isEmailVerified;
+  //     });
+  //   }
+  // }
+
+  @override
+  void initState() {
+    super.initState();
+    Timer _timer;
+    // code for auto-check
+
+    Future(() async {
+      _timer = Timer.periodic(Duration(seconds: 5), (timer) async {
+        FirebaseUser olduser = await FirebaseAuth.instance.currentUser();
+        await olduser.reload();
+        var user = await FirebaseAuth.instance.currentUser();
+        if (user.isEmailVerified) {
+          setState(() {
+            verified = user.isEmailVerified;
+          });
+          timer.cancel();
+        }
+      });
+    });
+    _timer = _timer;
```

Comment:
What does this line achieve?

Reply:
Maybe mention in the comments what you are trying to do.

Reply:
Flutter analyze was showing an error that _timer isn't being used anywhere, idk why. Hence this line to bypass that :p

Reply:
If you're not using the returned value you could just remove the _timer =
Just treat it as no return statement.

Reply:
fixed, will push it along next set of feature additions.